products/BellHybrid/apps/application-bell-relaxation/model/RelaxationSongsRepository.cpp

```diff
@@ -0,0 +1,135 @@
+            updateFilesCount(musicType, musicPath);
+        }
+    }
+
+    void RelaxationSongsRepository::updateFilesCount(const MusicType &type, const std::string &path)
+    {
+        auto query = std::make_unique<db::multimedia_files::query::GetCountForPath>(path);
+        auto task  = app::AsyncQuery::createFromQuery(std::move(query), db::Interface::Name::MultimediaFiles);
+        task->setCallback([this, &type](auto response) {
+            auto result = dynamic_cast<db::multimedia_files::query::GetCountResult *>(response);
+            if (result == nullptr) {
+                return false;
+            }
+            return updateCountCallback(type, result->getCount());
+        });
+        task->execute(application, this);
+    }
+
+    bool RelaxationSongsRepository::updateCountCallback(const MusicType &type, std::uint32_t count)
+    {
+        filesPerType[type] = count;
+        return true;
+    }
+
+    void RelaxationSongsRepository::getMusicFiles(
+        const MusicType &musicType,
+        std::uint32_t offset,
+        std::uint32_t limit,
+        const RelaxationSongsRepository::OnGetMusicFilesListCallback &viewUpdateCallback)
+    {
+        auto taskCallback = [this, viewUpdateCallback, offset, musicType](auto response) {
+            auto result = dynamic_cast<db::multimedia_files::query::GetLimitedResult *>(response);
+            if (result == nullptr) {
+                return false;
+            }
+            for (auto &record : result->getResult()) {
+                musicFilesViewCache.records.push_back(record);
+            }
+            musicFilesViewCache.recordsOffset = offset;
+            musicFilesViewCache.recordsCount  = getRecordsCount();
+
+            if (viewUpdateCallback) {
+                viewUpdateCallback(musicFilesViewCache.records, musicFilesViewCache.recordsCount);
+            }
+            updateCountCallback(musicType, result->getCount());
+            return true;
+        };
+
+        auto query = std::make_unique<db::multimedia_files::query::GetLimitedByPaths>(
+            std::vector<std::string>{pathPrefixes[musicType]}, offset, limit);
+        auto task = app::AsyncQuery::createFromQuery(std::move(query), db::Interface::Name::MultimediaFiles);
+        task->setCallback(taskCallback);
+        task->execute(application, this);
+    }
+
+    std::uint32_t RelaxationSongsRepository::getRecordsCount()
+    {
+        std::uint32_t sum{0};
+        std::for_each(
+            filesPerType.begin(), filesPerType.end(), [&sum](const auto &record) mutable { sum += record.second; });
+        return sum;
+    }
+
+    void RelaxationSongsRepository::getMusicFiles(
+        std::uint32_t offset,
+        std::uint32_t limit,
+        const RelaxationSongsRepository::OnGetMusicFilesListCallback &callback)
+    {
+        const auto skipViewUpdate = [](const std::vector<db::multimedia_files::MultimediaFilesRecord> &, unsigned int) {
```

Comment:
IMO it's better to name all parameters in functions and just explicitly mark them as `[[maybe_unused]]`, you can shorten the arguments declaration by using generic lambda `[](const auto &records, const auto count)`